### PR TITLE
Fix lag setting state in settings

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205056561538274/f

### Description
Fix a lag in setting the NetP state when coming into settings

### Steps to test this PR
Verify repro steps in [asana task](https://app.asana.com/0/488551667048375/1205056561538274/f) don't repro anymore

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3347** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3347" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3347?utm_source=stack-comment).

File: app-tracking-protection/app-tracking-api/src/main/java/com/duckduckgo/mobile/android/app/tracking/AppTrackingProtection.kt [hunk 1]
```diff
@@ -17,5 +17,24 @@
 package com.duckduckgo.mobile.android.app.tracking
 
 interface AppTrackingProtection {
+
+    /**
+     * This method returns whether the user has gone through AppTP onboarding
+     * @return Returns `true` if user has gone through AppTP onboarding, `false` otherwise
+     */
```
Comment (aitorvs) [inline app-tracking-protection/app-tracking-api/src/main/java/com/duckduckgo/mobile/android/app/tracking/AppTrackingProtection.kt:24]: Incidentally adding missing java docs

File: app/src/main/res/values/donottranslate.xml [hunk 2]
```diff
@@ -27,7 +27,6 @@
     <!-- Network Protection -->
     <string name="netpSettingsTitle">Network Protection</string>
     <string name="netpSettingsNeverEnabled">Secure your network connection</string>
-    <string name="netpSettingsConnecting">Connecting</string>
```
Comment (aitorvs) [inline app/src/main/res/values/donottranslate.xml:30]: no longer used

File: app/src/main/java/com/duckduckgo/app/settings/SettingsViewModel.kt [hunk 3]
```diff
@@ -81,16 +77,14 @@ class SettingsViewModel @Inject constructor(
...
+        val networkProtectionStateEnabled: Boolean = false,
         val networkProtectionWaitlistState: NetPWaitlistState = NetPWaitlistState.NotUnlocked,
         val isAutoconsentEnabled: Boolean = false,
     )
 
     enum class NetPState {
-        CONNECTING,
         CONNECTED,
         DISCONNECTED,
-        INVALID,
```
Comment (aitorvs) [inline app/src/main/java/com/duckduckgo/app/settings/SettingsViewModel.kt:93]: Connecting and invalid are no longer used in the new settings page

**Review by karlenDimla — APPROVED** [review]

**Pull request merged**